### ISSUE TITLE
`show_whitespace` highlighter overrides every other highlight

### ISSUE THREAD

**Issue by occivink** [mawww/kakoune#1251]
Easily reproducible with this snippet:

```
hook global WinCreate .* %{
  addhl show_whitespaces
  addhl dynregex '%reg{/}' 0:blue,red
}
```

to reproduce, set the `/` register to any pattern containing whitespace. The `Whitespace` will take precedence on whitespace characters. 

I would expect that when the dynregex highlighter is added after the other, the highlighter also gets applied after.

**Comment by occivink** [comment]
Kinda fixed by 777ba287e18f2f2452bd90a02f694395e839ff31 but to work you have to declare the dynregex before, so like that:
```
hook global WinCreate .* %{
  addhl dynregex '%reg{/}' 0:blue,red
  addhl show_whitespaces
}
```
whereas other highlighters get applied in the order of declaration.

**Comment by bhajneet** [comment]
Thank you for this, it helped me figure out why I couldn't get my highlighter regex to work.
